src/openloco/windows/industry_list.cpp

```diff
@@ -1,16 +1,1118 @@
+            auto args = FormatArguments();
+            args.push(industryCost);
+
+            auto widthOffset = 0;
+            if (!is_editor_mode())
+            {
+                auto xPos = self->x + 3 + self->width - 19;
+                auto yPos = self->y + self->height - 13;
+                widthOffset = 138;
+                gfx::draw_string_494C78(*dpi, xPos, yPos, colour::black, string_ids::build_cost, &args);
+            }
+            auto xPos = self->x + 3;
+            auto yPos = self->y + self->height - 13;
+            auto width = self->width - 19 - widthOffset;
+            gfx::draw_string_494BBF(*dpi, xPos, yPos, width, colour::black, string_ids::white_stringid2);
+        }
+
+        // 0x0045843A
+        static void on_mouse_up(window* self, widget_index widgetIndex)
+        {
+            switch (widgetIndex)
+            {
+                case common::widx::close_button:
+                    WindowManager::close(self);
+                    break;
+
+                case common::widx::tab_industry_list:
+                case common::widx::tab_new_industry:
+                    common::switchTab(self, widgetIndex);
+                    break;
+            }
+        }
+
+        //0x00458966
+        static void on_scroll_mouse_down(ui::window* self, int16_t x, int16_t y, uint8_t scroll_index)
+        {
+            std::printf("currentTabOnScrollMouseDown: %d\n", self->current_tab);
+            registers regs;
+            regs.esi = (int32_t)self;
+            regs.cx = x;
+            regs.dx = y;
+            call(0x00458966, regs);
+        }
+
+        // 0x00458721
+        static void on_scroll_mouse_over(ui::window* self, int16_t x, int16_t y, uint8_t scroll_index)
+        {
+            std::printf("currentTabOnScrollMouseOver: %d\n", self->current_tab);
+            registers regs;
+            regs.esi = (int32_t)self;
+            regs.cx = x;
+            regs.dx = y;
+            call(0x00458721, regs);
+        }
+
+        // 0x004585B8
+        static void on_update(window* self)
+        {
+            std::printf("currentTabOnUpdate: %d\n", self->current_tab);
+            if ((_screen_flags & screen_flags::unknown_5) != 0)
+            {
+                auto cursor = input::getMouseLocation();
+                auto xPos = cursor.x;
+                auto yPos = cursor.y;
+                window* activeWindow = WindowManager::findAt(xPos, yPos);
+                if (activeWindow == self)
+                {
+                    xPos -= self->x;
+                    xPos += 26;
+                    yPos -= self->y;
+
+                    if (yPos >= 42)
+                    {
+                        xPos = cursor.x;
+                        yPos = cursor.y;
+                        widget_index activeWidget = self->find_widget_at(xPos, yPos);
+                        if (activeWidget == common::widx::panel)
+                        {
+                            self->saved_view.mapX += 1;
```

Comment:
saved_view.mapX looks to not be valid for this window. Its likely we will have to create a union for this field as saved_views only make sense with windows with a viewport which this of course does not have.